### ISSUE TITLE
TypeError with client-dynamodb 

### ISSUE THREAD

**Issue by garrettheel** [aws/aws-sdk-js-v3#1167]
**Describe the bug**

Following on from https://github.com/aws/aws-sdk-js-v3/issues/1140, it seems like master (unreleased; 6a447a0fc559e6e7f0cb9dcae71997f3c62f5ef7) has some changes that might address that issue, so I built from source to test.

That results in the following new exception:

```
(node:28876) UnhandledPromiseRejectionWarning: TypeError: context.endpoint is not a function
    at buildHttpRpcRequest (/Users/gheel/src/dynamodb-admin/node_modules/@aws-sdk/client-dynamodb/dist/cjs/protocols/Aws_json1_0.js:6240:66)
    at Object.exports.serializeAws_json1_0ListTablesCommand (/Users/gheel/src/dynamodb-admin/node_modules/@aws-sdk/client-dynamodb/dist/cjs/protocols/Aws_json1_0.js:187:12)
    at serialize (/Users/gheel/src/dynamodb-admin/node_modules/@aws-sdk/client-dynamodb/dist/cjs/commands/ListTablesCommand.js:25:30)
    at Object.<anonymous> (/Users/gheel/src/dynamodb-admin/node_modules/@aws-sdk/middleware-serde/build/serializerMiddleware.js:7:31)
    at Generator.next (<anonymous>)
    at fulfilled (/Users/gheel/src/dynamodb-admin/node_modules/tslib/tslib.js:110:62)
```

**SDK version number**

Unreleased, testing from master @ 6a447a0fc559e6e7f0cb9dcae71997f3c62f5ef7

**Is the issue in the browser/Node.js?**
Node.js

**Details of the browser/Node.js version**
Paste output of `npx envinfo --browsers` or `node -v`

```
$ node -v
v10.20.1
```

**To Reproduce (observed behavior)**

```node
(async () => {
  AWS = require("@aws-sdk/client-dynamodb");
  client = new AWS.DynamoDB({
    tls: false,
    endpoint: "http://localhost:8000",
  });
  await client.listTables({});  // raises error
})();
```

Exception:

```
(node:28876) UnhandledPromiseRejectionWarning: TypeError: context.endpoint is not a function
    at buildHttpRpcRequest (/Users/gheel/src/dynamodb-admin/node_modules/@aws-sdk/client-dynamodb/dist/cjs/protocols/Aws_json1_0.js:6240:66)
    at Object.exports.serializeAws_json1_0ListTablesCommand (/Users/gheel/src/dynamodb-admin/node_modules/@aws-sdk/client-dynamodb/dist/cjs/protocols/Aws_json1_0.js:187:12)
    at serialize (/Users/gheel/src/dynamodb-admin/node_modules/@aws-sdk/client-dynamodb/dist/cjs/commands/ListTablesCommand.js:25:30)
    at Object.<anonymous> (/Users/gheel/src/dynamodb-admin/node_modules/@aws-sdk/middleware-serde/build/serializerMiddleware.js:7:31)
    at Generator.next (<anonymous>)
    at fulfilled (/Users/gheel/src/dynamodb-admin/node_modules/tslib/tslib.js:110:62)
```

Note that `context.endpoint` looks to already be resolved, and is:
```
  endpoint:
   { hostname: 'localhost',
     port: 8000,
     protocol: 'http:',
     path: '/',
     query: undefined },
```

**Expected behavior**
Should make a request using the http protocol.

**Screenshots**
N/A

**Additional context**
N/A

**Comment by AllanFly120** [comment]
Looks like the commits from PR: https://github.com/aws/aws-sdk-js-v3/pull/1106 got overwritten for some reason, we will investigate the root cause.

**Comment by AllanFly120** [comment]
@garrettheel Can you clone the master and try again? Please make sure you run `git clean -xdf` before building from source. I think it should be good now.

**Comment by matej-prokop** [comment]
@AllanFly120 Would it be possible to release a new beta version including this bugfix?

**Comment by fboudra** [comment]
still broken on 1.0.0-gamma.2

**Comment by trivikr** [comment]
Closing as issue is fixed in 1.0.0-gamma.2
